Doc/whatsnew/3.7.rst

```diff
@@ -638,10 +638,18 @@ Changes in Python behavior
 
       f(1 for x in [1],)
 
+      @deco(1 for x in [1])
```

Comment:
What's wrong with this one? While we've restricted the form that decorator expressions take to being a name (dotted or otherwise) plus an optional function call, we've never restricted the arguments passed to that function call.

At the generator expression level, it also still meets the "must be surrounded by parentheses rule".

Reply:
The same as with
```
@deco[0](1)
```
or
```
@deco(0)(1)
```
or
```
@(a+b*c)
```
This looks like valid Python expression syntax, but the syntax of decorator expression is intentionally more limited.

Reply:
This aspect of the change is more akin to prohibiting `@deco(a+b*c)` - once the top level expression is in the form of a function call, the fact that it's part of a decorator expression shouldn't be altering what's permitted in the argument list.

Reply:
In any case these syntax changes should be discussed and officially approved. Could you please open a topic on Python-Dev about omitting parenthesis in a generator expression in these two cases? I have arguments for these changes and against them, but I think you will formulate them better. I'll add my comments.

Despite the fact that currently the CPython parser allows such syntax, it is illegal, and allowing it officially is a changing of Python language specification.